src/hera/workflows/workflow.py

```diff
@@ -330,13 +332,57 @@ def to_yaml(self, *args, **kwargs) -> str:
         kwargs.setdefault("sort_keys", False)
         return _yaml.dump(self.to_dict(), *args, **kwargs)
 
-    def create(self) -> TWorkflow:
-        """Creates the Workflow on the Argo cluster."""
+    def create(self, wait: bool = False, poll_interval: int = 5) -> TWorkflow:
+        """Creates the Workflow on the Argo cluster.
+
+        Parameters
+        ----------
+        wait: bool = False
+            If true then the workflow is created asynchronously and the function returns immediately.
+            If false then the workflow is created and the function blocks until the workflow is done executing.
+        poll_interval: int = 5
+            The interval in seconds to poll the workflow status if wait is true. Ignored when wait is true.
+        """
         assert self.workflows_service, "workflow service not initialized"
         assert self.namespace, "workflow namespace not defined"
-        return self.workflows_service.create_workflow(
+
+        wf = self.workflows_service.create_workflow(
             WorkflowCreateRequest(workflow=self.build()), namespace=self.namespace
         )
+        # set the workflow name to the name returned by the API, which helps cover the case of users relying on
+        # `generate_name=True`
+        self.name = wf.metadata.name
+
+        if wait:
+            return self.wait(poll_interval=poll_interval)
+        return wf
+
+    def wait(self, poll_interval: int = 5) -> TWorkflow:
+        """Waits for the Workflow to complete execution.
+
+        Parameters
+        ----------
+        poll_interval: int = 5
+            The interval in seconds to poll the workflow status.
+        """
+        assert self.workflows_service is not None, "workflow service not initialized"
+        assert self.namespace is not None, "workflow namespace not defined"
+        assert self.name is not None, "workflow name not defined"
+
+        wf = self.workflows_service.get_workflow(self.name, namespace=self.namespace)
+        assert wf.status is not None, f"workflow status not defined for workflow {wf.metadata.name}"
+        assert wf.status.phase is not None, f"workflow phase not defined for workflow status {wf.status}"
+        status = WorkflowStatus.from_argo_status(wf.status.phase)
+        assert wf.metadata.name is not None, f"workflow name not defined for workflow {self.name}"
```

Comment:
Think it makes more sense to have the name assert first as the status assert uses it. Added a whitespace for readability
```suggestion
        wf = self.workflows_service.get_workflow(self.name, namespace=self.namespace)
        assert wf.metadata.name is not None, f"workflow name not defined for workflow {self.name}"

        assert wf.status is not None, f"workflow status not defined for workflow {wf.metadata.name}"
        assert wf.status.phase is not None, f"workflow phase not defined for workflow status {wf.status}"
        status = WorkflowStatus.from_argo_status(wf.status.phase)
```